### PR TITLE
[release-1.20] - Backport K3s/RKE2 bootstrap data is encrypted with empty string if user does not supply a token

### PR DESCRIPTION
**Pull request by galal-hussein**
#### Linked Issues ####

* https://github.com/rancherlabs/rancher-security/issues/832
* #3630 
* #3631

### CODE REVIEW

**Comment by brandond** [comment]
Will need https://github.com/k3s-io/k3s/pull/3514 rolled in as well.

**Review by brandond — CHANGES_REQUESTED** [review]
Will need https://github.com/k3s-io/k3s/pull/3545 rolled in as well.

**Review by briandowns — APPROVED** [review]

**Review by brandond — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -86,9 +86,9 @@ require (
 	github.com/google/uuid v1.1.2
 	github.com/gorilla/mux v1.8.0
 	github.com/gorilla/websocket v1.4.2
-	github.com/k3s-io/helm-controller v0.8.3
-	github.com/k3s-io/kine v0.6.0
-	github.com/klauspost/compress v1.11.7
+	github.com/k3s-io/helm-controller v0.9.1
+	github.com/k3s-io/kine v0.6.1
```
Comment (brandond) [inline go.mod:90]: Need to pull in https://github.com/k3s-io/k3s/commit/246b378a2743e659a2343b60123ba51a38dcc0ac for kine delete fix

**Review by brandond — CHANGES_REQUESTED** [review]

**Comment by galal-hussein** [comment]
https://github.com/k3s-io/k3s/issues/3630 adding issue for backporting updating kine to fix delete issue

**Review by brandond — APPROVED** [review]

**Comment by galal-hussein** [comment]
https://github.com/k3s-io/k3s/issues/3631 - backport issue for empty node password file

**Pull request merged**